proto/caraml/upi/v1/upi.proto

```diff
@@ -19,10 +20,22 @@ service UniversalPredictionService {
 
 // Represents a request to predict multiple values
 message PredictValuesRequest {
-  // Collection of prediction instances to be predicted.
-  // Each prediction row correspond to one prediction instance.
-  // NOTE: the ordering of prediction_rows might differ with prediction_result_rows in the response
-  repeated PredictionRow prediction_rows = 1;
+  // Prediction table contains instances to be predicted.
+  // Each row in the table correspond to one prediction instance.
+  // Prediction table should contain all preprocessed feature that model use to perform prediction.
+  // The column ordering in the prediction table must be the same as feature order expected by model in the case of standard model.
+  // Prediction table can be populated via 3 ways:
+  // - By performing preprocessing in the client-side and sent as part of original request.
+  // - By transforming feature values stored in transformer_inputs.
+  // - By retrieving precomputed feature value from feature store. 
+  // NOTE: the ordering of rows might differ in the response but the number of row must remain the same.
+  Table prediction_table = 1;
```

Comment:
can this be empty in the original request (with all data passed in `transformer_input`)? or does the user always need to define row_ids here?

Reply:
`prediction_table` is mandatory for standard model case. But then nothing prevent them doing that when transformer and pyfunc are involved, especially if we take the route of making prediction log as part of standard transformer operation. In such case, one option to enforce it is to have client library that wraps the UPI interface, and the library become the primary interface that client uses.

Reply:
got it. could consider using https://github.com/envoyproxy/protoc-gen-validate

Reply:
Ok, will have a look into it.